src/server/ftpserver.rs

```diff
@@ -280,11 +484,13 @@ where
         let storage = Arc::new((self.storage)());
         let storage_features = storage.supported_features();
         let authenticator = self.authenticator.clone();
-        let session = Session::new(storage)
+        let mut session = Session::new(storage)
             .ftps(self.certs_file.clone(), self.certs_password.clone())
             .metrics(with_metrics);
+        let (control_msg_tx, control_msg_rx): (Sender<InternalMsg>, Receiver<InternalMsg>) = channel(1);
+        session.control_msg_tx = Some(control_msg_tx.clone());
+        session.control_connection = control_connection;
```

Comment:
Perhaps we can just rename `control_connection` to `control_connection` info since it makes one think that this is a stream while it's just information regarding the control connection. Thoughts?

Reply:
thought = i totally agree. i'll push a change for this